frontend/CONTRIBUTING.md

```diff
@@ -0,0 +1,271 @@
+
+This CONTRIBUTING.md is a javascript/typescript web app using Bun as a replacement for Node.js (a non-browser JavaScript runtime environment) with the Vite/Vue framework. You can use the monarch-app frontend to query the Monarch KG using the web app locally.
+
+Without additional setup this will use the same FastAPI backend and data source as the website. This is primarily for front-end development of the website UI. If you want to use a local data store review the monarch-app README.md and CONTRIBUTING.md in the top level directory of this repository to understand how to manage the whole stack locally.
+
+# Useful Links
+
+- [Monarch App Website](https://next.monarchinitiative.org/) - The Monarch Initiative website, a Vue webapp accessing `monarch-py`, a Python library for interacting with the Monarch Initiative knowledge graph, which includes an optional FastAPI module that serves as the website's backend.
+- [Documentation](https://monarch-initiative.github.io/monarch-documentation/) - In addition, to the documentation here, we have a separate documentation site that is automatically generated from the codebase.
+
+# Table of Contents
+
+- [Welcome to the Monarch Initiative](#welcome-to-the-monarch-initiative)
+- [Useful Links](#useful-links)
+- [Quick Start](#quick-start)
+  - [Dependancies](#dependancies)
+- [Monarch Frontend Details](#monarch-frontend-details)
+  - [Commands](#commands)
+  - [Flags](#flags)
+  - [Phenogrid](#phenogrid)
+    - [Parameters](#parameters)
+    - [Events](#events)
+  - [Style guidelines](#style-guidelines)
+
+# Quick Start
+
+To get started running or testing the frontend environment, install the dependancies then
+
+## Dependancies
+
+- [Bun](bun.sh)
+
+Bun is used as a drop-in replacement for Node (non-browser JavaScript runtime environment) and Yarn (package manager).
+It is _not_ yet used as a replacement for Vite (dev previewing, bundling, building) or Vitest (test runner).
+
+This makes one less thing for developers to install.
+It also should make installs (which impact GitHub Actions quota) much faster, and anything that runs locally (like Vite and Vitest) a bit faster as well.
+
+Anything that works in Node should also work in Bun, per Bun's stated goals.
+Do not use Bun-specific APIs.
+If there is ever a problem with Bun, switching back to Node should be as simple as replacing "bun" with "node"/"yarn" and "bunx" with "npx" in commands.
+
+To install Bun with an existing Node.js environment (currently recommended for version/package isolation):
+
+```
+npm install -g bun
+```
+
+If you don't have an existing Node.js implementation and want to install Bun globally on your system (requires curl):
+
+```
+curl -fsSL https://bun.sh/install | bash
+```
+
+## Run using Monarch Cloud hosted API
+
+To run the frontend using the Monarch Cloud hosted API, you can use the following commands:
```

Comment:
Might need to explicitly mention being in the frontend directory  (or imply it with a little fake prompt?)

Reply:
ohh, never mind, this is the readme within the frontend directory!